templates/commands/render.go

```diff
@@ -429,10 +423,26 @@ func copyRecursive(pos *model.ConfigPos, from, to string, rfs renderFS, overwrit
 			return fmt.Errorf("Stat(): %w", err)
 		}
 
+		rf, err := rfs.Open(path)
+		if err != nil {
+			return model.ErrWithPos(pos, "Open(): %w", err) //nolint:wrapcheck
+		}
+		defer func() { outErr = errors.Join(outErr, rf.Close()) }()
+
+		if dryRun {
+			return nil
+		}
+
 		// The permission bits on the output file are copied from the input file;
 		// this preserves the execute bit on executable files.
-		if err := rfs.WriteFile(dst, buf, info.Mode().Perm()); err != nil {
-			return fmt.Errorf("failed writing to scratch file: WriteFile(): %w", err)
+		wf, err := rfs.OpenFile(dst, os.O_CREATE|os.O_WRONLY, info.Mode().Perm())
+		if err != nil {
+			return model.ErrWithPos(pos, "OpenFile(): %w", err) //nolint:wrapcheck
+		}
+		defer func() { outErr = errors.Join(outErr, wf.Close()) }()
+
+		if _, err := io.Copy(wf, rf); err != nil {
+			return fmt.Errorf("Copy(): %w", err)
```

Comment:
nit: it seems it's not consistent when we will add path or not. On line 418, we have `Stat(%s)`. On line 423 we have `Stat()`. Was that intentional or we should make them consistent?

Reply:
Fixed, stat() includes its path already, so we don't need the %s.